File: maui/KsDialogs.Maui.ApiSurfaceCheck/NegativeChecks/RejectsLayoutAttributeOnViewModel.cs
namespace KsDialogs.ApiSurfaceCheck.NegativeChecks;

/// <summary>ViewModel はレイアウト属性を持たない (core/ADR-0014)。</summary>
/// <remarks>
/// このソースはビルドプロパティ KsDialogsNegativeCheckVmAttribute を true にしたときだけ加わり、
/// <b>コンパイルエラーで失敗すること</b>が期待結果になる。
/// 期待する診断: CS1061 ('ConsumerDialogViewModel' に 'ProportionalWidth' の定義が含まれていない)
/// </remarks>
internal static class RejectsLayoutAttributeOnViewModel
{
    public static double Read() => new ConsumerDialogViewModel().ProportionalWidth;
}

File: android/api-surface-check/src/negativeCheckVmAttribute/kotlin/jp/kamusoft/ksdialogs/apicheck/RejectsLayoutAttributeOnViewModel.kt
package jp.kamusoft.ksdialogs.apicheck

/**
 * ViewModel はレイアウト属性を持たない (core/ADR-0014)。
 *
 * このソースは `-Pksdialogs.negativeCheck.vmAttribute` を付けたときだけビルドに加わり、
 * **コンパイルエラーで失敗すること**が期待結果になる。
 * 期待する診断: Unresolved reference 'proportionalWidth'.
 */
public object RejectsLayoutAttributeOnViewModel {
    public fun read(): Double = ConsumerDialogViewModel("こんにちは").proportionalWidth
}
